guides/source/autoloading_and_reloading_constants.md

```diff
@@ -185,6 +184,15 @@ if `User` is reloaded, since `VipUser` is not, the superclass of `VipUser` is th
 
 Bottom line: **do not cache reloadable classes or modules**.
 
+When classes do need to be autoloaded, we have an initializer just for that. The callback will run once on application startup and every time code is reloaded.
```

Comment:
Techincally, this is not really an initializer. Better say it is a reloader callback. Also, very important to say that for obscure reasons this callback runs twice, so the content of the block has to be idempotent.

Reply:
I'm curious, what obscure reasons are they? Where can I read more about this?

Reply:
It is undocumented I believe. These callbacks had to maintain a delicate equilibrium with past implementations and different runtime modes. @matthewd has the whole story, but for the purposes of this guide that is too deep and private. We may also say "historical" instead of "obscure".

Reply:
> But for the purposes of this guide that is too deep and private.

👍 was curious how it works for my understanding 🙂